src/components/Dialog/Dialog.js

```diff
@@ -26,15 +26,28 @@ function Dialog(props) {
   const dispatch = useDispatch();
   const { body, isModal } = props;
 
-  const escFunction = () => {
-    dispatch(setDialogVisible());
-  };
+  const dialogRef = useRef(null);
+  let activeElement;
+
+  const escFunction = e => e.which === 27 && dispatch(setDialogVisible());
 
   const registerEsc = isRegister => {
```

Comment:
Since you use this function only once, I will move the content inside the useEffect directly.
This function will be overriden on each render, I think it's not good. Same for `activeElement` on each render the variable will be set to undefined so define it inside the useEffect. 

The code works because we mount/unmount the component and never rerender it .